app/models/form1095_b.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+class Form1095B < ApplicationRecord
+  has_kms_key
+  encrypts :form_data, key: :kms_key, **lockbox_options
+
+  # validations
+  validates :veteran_icn, :tax_year,  presence: true
+  validates :veteran_icn, uniqueness: { scope: :tax_year }
+  validate :proper_form_data_schema
+
+  # scopes
+  scope :get_available_forms, lambda { |icn|
+    where(veteran_icn: icn).order(tax_year: :desc).pluck(:tax_year, :updated_at)
+  }
+
+  # methods
+  def get_pdf
+    generate_pdf
+  end
+
+  private
+
+  def country_and_zip
+    "#{data['country']} #{data['zip_code'] || data['foreign_zip']}"
+  end
+
+  def middle_initial
+    data['middle_name'] ? data['middle_name'][0] : ''
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def generate_pdf
```

Comment:
This method is quite long. Could you move some of this logic to the `pdf` method? Thoughts?
Something like:

```ruby 
def pdf 
  pdftk = PdfForms.new(Settings.binaries.pdftk)

  tmp_file = Tempfile.new("1095B-#{SecureRandom.hex}.pdf")
  pdf_template = "lib/pdf_fill/forms/pdfs/f1095bs/1095b-#{tax_year}.pdf"

  unless File.exist?(pdf_template)
    Rails.logger.error "1095-B template for year #{tax_year} does not exist."
    raise "1095-B for tax year #{tax_year} not supported"
  end

  generate_pdf(pdftk, tmp_file, pdf_template)

raise PdfForms::PdftkError => e
  ...
end


private 

def generate_pdf(pdftk, tmp_file, pdf_template)
  pdftk.fill_form(pdf_template, tmp_file, { ... }, flatten: true)

  pdf = temp_file.read
  temp_file.close
  temp_file.unlink

  pdf
end
```

Reply:
Yeah i like that